src/data/patch.ts

```diff
@@ -0,0 +1,357 @@
+import {type Observable} from 'rxjs'
+
+import type {ObservableSanityClient, SanityClient} from '../SanityClient'
+import type {
+  AllDocumentIdsMutationOptions,
+  AllDocumentsMutationOptions,
+  AttributeSet,
+  BaseMutationOptions,
+  FirstDocumentIdMutationOptions,
+  FirstDocumentMutationOptions,
+  FIXME,
+  MultipleMutationResult,
+  PatchMutationOperation,
+  PatchOperations,
+  PatchSelection,
+  SanityDocument,
+  SingleMutationResult,
+} from '../types'
+import getSelection from '../util/getSelection'
+import {validateInsert, validateObject} from '../validators'
+
+/** @internal */
+export class BasePatch {
+  protected selection: PatchSelection
+  protected operations: PatchOperations
+  constructor(selection: PatchSelection, operations: PatchOperations = {}) {
+    this.selection = selection
+    this.operations = operations
+  }
+
+  /**
+   * DEPRECATED: Don't use.
+   * The operation is added to the current patch, ready to be commited by `commit()`
+   *
+   * @deprecated - Don't use.
+   * @param attrs - Attributes to replace
+   */
+  replace(attrs: AttributeSet): this {
+    validateObject('replace', attrs)
+    return this._set('set', {$: attrs})
+  }
```

Comment:
Given this has been deprecated for a while, should we remove it?

Reply:
We can, all the remaining deprecations are there because I couldn't find, or didn't know, how to explain the before/after migration for it. Do you have one for this?

Reply:
The rationale is that it is unclear which properties this actually replaces in a document. What you usually want is either the `set()` patch (replacing the passed properties), or `createOrReplace()` (replacing the entire document)